From f9bccdab8676837dc5c4b9313f5342ea2b73dbe0 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:26 -0700
Subject: [PATCH 1651/1852] v4l: xilinx: remapper: Fix warnings for
 NULL_RETURNS

commit f364f105fe9829c007022ea56d1ac6064c580711 from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the NULL_RETURNS type.

The actual warnings are:

For xremap_get_pad_format():
Event return_null: 	Explicitly returning null.

For xremap_get_format():
Event returned_null: 	"xremap_get_pad_format" returns "NULL".
Event dereference: 	Dereferencing "xremap_get_pad_format(xremap,
cfg, fmt->pad, fmt->which)", which is known to be "NULL".

For xremap_set_format():
Event returned_null: 	"xremap_get_pad_format" returns "NULL"
Event var_assigned: 	Assigning: "format" = "NULL" return value from
"xremap_get_pad_format"
Event dereference: 	Dereferencing "format", which is known to be
"NULL".

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../media/platform/xilinx/xilinx-remapper.c   | 26 +++++++++++++++----
 1 file changed, 21 insertions(+), 5 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-remapper.c b/drivers/media/platform/xilinx/xilinx-remapper.c
index cfda110d1432..66ce55204414 100644
--- a/drivers/media/platform/xilinx/xilinx-remapper.c
+++ b/drivers/media/platform/xilinx/xilinx-remapper.c
@@ -218,15 +218,22 @@ xremap_get_pad_format(struct xremap_device *xremap,
 		      struct v4l2_subdev_pad_config *cfg,
 		      unsigned int pad, u32 which)
 {
+	struct v4l2_mbus_framefmt *format;
+
 	switch (which) {
 	case V4L2_SUBDEV_FORMAT_TRY:
-		return v4l2_subdev_get_try_format(&xremap->xvip.subdev, cfg,
-						  pad);
+		format = v4l2_subdev_get_try_format(&xremap->xvip.subdev, cfg,
+						    pad);
+		break;
 	case V4L2_SUBDEV_FORMAT_ACTIVE:
-		return &xremap->formats[pad];
+		format = &xremap->formats[pad];
+		break;
 	default:
-		return NULL;
+		format = NULL;
+		break;
 	}
+
+	return format;
 }
 
 static int xremap_get_format(struct v4l2_subdev *subdev,
@@ -234,8 +241,13 @@ static int xremap_get_format(struct v4l2_subdev *subdev,
 			     struct v4l2_subdev_format *fmt)
 {
 	struct xremap_device *xremap = to_remap(subdev);
+	struct v4l2_mbus_framefmt *format;
 
-	fmt->format = *xremap_get_pad_format(xremap, cfg, fmt->pad, fmt->which);
+	format = xremap_get_pad_format(xremap, cfg, fmt->pad, fmt->which);
+	if (!format)
+		return -EINVAL;
+
+	fmt->format = *format;
 
 	return 0;
 }
@@ -251,6 +263,8 @@ static int xremap_set_format(struct v4l2_subdev *subdev,
 	unsigned int i;
 
 	format = xremap_get_pad_format(xremap, cfg, fmt->pad, fmt->which);
+	if (!format)
+		return -EINVAL;
 
 	if (fmt->pad == XREMAP_PAD_SOURCE) {
 		fmt->format = *format;
@@ -288,6 +302,8 @@ static int xremap_set_format(struct v4l2_subdev *subdev,
 	/* Propagate the format to the source pad. */
 	format = xremap_get_pad_format(xremap, cfg, XREMAP_PAD_SOURCE,
 				       fmt->which);
+	if (!format)
+		return -EINVAL;
 	*format = fmt->format;
 	format->code = output->code;
 
-- 
2.31.1

